src/node.rs

```diff
@@ -173,6 +196,19 @@ impl<T> MutableNode for Node<T> {
         self.children.push(child);
     }
 
+    fn replace_child(&mut self, old: NodeId, new: NodeId) {
+        let mut index = 0;
+
+        for (i, id) in self.children().iter().enumerate()
+        {
+            if id == &old { index = i; break; }
+        }
```

Comment:
I believe something like this would also work:
```
let index = self.children()
    .iter()
    .enumerate()
    .find(|(i, id)| id == &old).0;
```
(Those references might not be quite right, of course).

Would you be against that implementation instead of the manual index + break?